src/main/java/org/junit/Assert.java

```diff
@@ -991,6 +996,8 @@ public static void assertThrows(Class<? extends Throwable> expectedThrowable, Th
     public static <T extends Throwable> T expectThrows(Class<T> expectedThrowable, ThrowingRunnable runnable) {
         try {
             runnable.run();
+        } catch (AssumptionViolatedException e) {
+            throw new AssumptionNotSupportedException("Assumptions are not supported in expectThrows/assertThrows", e);
```

Comment:
This would break tests that are explicitly asserting that a given lambda throws `AssumptionViolatedException`. This is a reasonable thing to test if you have a common testing library that you want to test:

    expectThrows(AssumptionViolatedException.class, () -> tester.verify());

I don't think these methods should special-case `AssumptionViolatedException`. The contracts for these methods are clear and simple, and the behavior you are adding would be unexpected to many users (even if documented). It also would make these methods differ from the JUnit5 versions.

Instead, I think `MultipleFailureException.assertEmpty()` should special case lists that contain one element of type `AssumptionViolatedException`.